### PR TITLE
Implements logical groups in yql queries

### PR DESCRIPTION
**Pull request by icarojerry**

### CODE REVIEW

**Comment by jprodrigues70** [comment]
@IcaroJerry  To the best of my knowledge, a folder with code style standards guide isn't usual. If you want that your code follow a pattern, I recommend PSR-12 that is the new convention defined to be used with PHP. And to do that, you can include [php_codesniffer](https://packagist.org/packages/squizlabs/php_codesniffer) in your composer.json require-dev
```
    "require-dev": {
        "squizlabs/php_codesniffer": "*"
    },

```

**Comment by icarojerry** [comment]
@jprodrigues70, thanks for your comments. But, that change should not be here. I will undo and make the correct pull request (https://github.com/Escavador/vespa-php/pull/3). And I will apply your suggestions there.

**Comment by jprodrigues70** [comment]
@IcaroJerry great!

File: src/Models/VespaYQLBuilder.php [hunk 1]
```diff
@@ -10,6 +10,9 @@ class VespaYQLBuilder
     public function __construct()
     {
         $this->search_condition_groups = [];
+        $this->last_group = &$this->search_condition_groups;
+        $new_group = &$this->createGroup($this->last_group);
+        $this->last_group = &$new_group;
```
Comment (gabrielpeixoto) [inline src/Models/VespaYQLBuilder.php]:
```suggestion
        $this->last_group= &$this->createGroup($this->search_condition_groups);
```

File: src/Models/VespaYQLBuilder.php [hunk 2]
```diff
@@ -31,35 +34,45 @@ public final function reset(): VespaYQLBuilder
...
     }
 
-    public function addStemmingCondition(string $term, bool $stemming = false, string $field = 'default', $operator = 'CONTAINS', $logical_operator = 'AND', $group_name = null): VespaYQLBuilder
+    public function addConditionGroup(\Closure $closure): VespaYQLBuilder
+    {
+        $last_group = &$this->last_group; // save the last group
+        $new_group = &$this->createGroup($this->last_group); // create a new groupś
+        $this->last_group = &$new_group;  // refresh the last group
+        $closure($this); // execute function with more clause in new group
+        $this->last_group = &$last_group; // put the old last group back
```
Comment (gabrielpeixoto) [inline src/Models/VespaYQLBuilder.php]: Aqui não esta sobrescrevendo o anterior não (`$this->last_group = &$new_group;`)?

**Comment by gabrielpeixoto** [comment]
I think the `last_group` variable should be called `current_group`

File: src/Models/VespaYQLBuilder.php [hunk 3]
```diff
@@ -10,16 +10,17 @@ class VespaYQLBuilder
     public function __construct()
     {
         $this->search_condition_groups = [];
+        $this->last_group = &$this->createGroup($this->search_condition_groups);
```
Comment (gabrielpeixoto) [inline src/Models/VespaYQLBuilder.php]: I think the last_group variable should be called current_group

**Pull request merged**